src/main/java/io/fabric8/jenkins/openshiftsync/ConfigMapClusterInformer.java

```diff
@@ -53,8 +53,8 @@ public void start() {
         LOGGER.info("Starting cluster wide configMap informer for {} !!" + namespaces);
         LOGGER.debug("listing ConfigMap resources");
         SharedInformerFactory factory = getInformerFactory();
-        Map<String, String> labels = singletonMap(IMAGESTREAM_AGENT_LABEL, IMAGESTREAM_AGENT_LABEL_VALUE);
-        OperationContext withLabels = new OperationContext().withLabels(labels);
+        Map<String, String[]> labels = singletonMap(IMAGESTREAM_AGENT_LABEL, IMAGESTREAM_AGENT_LABEL_VALUES);
```

Comment:
Yeah @akram I suspect this label combination is not achieving the desired result of supporting either label (maybe it is requiring both labels instead) and this is why the existing configmap watch e2e test is failing 

My guess is you'll need to create 2 informers, one for each label individually

Reply:
`kubectl` and also the `informer`  are able to watch on multiple labels values. This change adds this, and the informer should react on any of them (`OR`)  .
I did run the test locally also before the end of year vacations and I was seeing this timeout happening as well and tried to figure out where it was from but with no success.

Reply:
yeah could be a discrepancy between this java  based version and kubectl

I would suggest looking at the test and mimicking what it does manually so you can debug

Reply:
> 

I actually misread the test logs @akram 

The test that failed was `TestSecretCredentialSyncAfterStartup` ... the new one Alice added as part of her last fix.

So your change here should have no bearing on secret watches, right?

The only possible common path is the PodTemplateUtils change maybe ... I'll point it out in a sec.

I'm also about to try the specific test locally.

Reply:
Though I'm still confused as `TestSecretCredentialSyncAfterStartup` does not even start a build (that hangs forever and then cancel

Maybe the tests were running in parallel and that output was from the config map test ??? ... did not think it did that

Yeah I'm going to run the configmap and secret credential tests individually against a cluster with your changes and see.

Reply:
Yeah not sure what happened with my copy/paste at https://github.com/openshift/jenkins-sync-plugin/pull/410#issuecomment-1004851896
but with the logs in the latest test run, it is TestConfigMapLegacyPodTemplate that failed

```
=== RUN   TestConfigMapLegacyPodTemplate
    sync_plugin_test.go:108: templateinstance jenkins-sync-plugin-test-namespace-ts2jp/jenkins-ephemeral instantiation ready
    sync_plugin_test.go:210: build phase Pending
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Running
    sync_plugin_test.go:210: build phase Cancelled
panic: test timed out after 1h15m0s
```

TestSecretCredentialSyncAfterStartup passed, and TestConfigMapLegacyPodTemplate failed for me locally.

The pod template from the config map is not created in the jenkins instance.

I do see these logs in the jenkins logs:

```
2022-01-05 18:28:09 INFO    io.fabric8.jenkins.openshiftsync.PodTemplateUtils purgeTemplates Purging PodTemplates for from Configmap with Uid 26a4f07a-5b95-4e31-ad42-37b834e1bcba
2022-01-05 18:28:09 INFO    io.fabric8.jenkins.openshiftsync.PodTemplateUtils removePodTemplate Removing PodTemplate: jenkins-slave in namespace:  null
2022-01-05 18:28:09 INFO    io.fabric8.jenkins.openshiftsync.PodTemplateUtils removePodTemplate Removing PodTemplate: jenkins-slave in namespace:  null
2022-01-05 18:28:09 INFO    io.fabric8.jenkins.openshiftsync.PodTemplateUtils removePodTemplate Removing PodTemplate: jenkins-slave in namespace:  null
2022-01-05 18:28:09 INFO    io.fabric8.jenkins.openshiftsync.PodTemplateUtils addPodTemplate Adding PodTemplate: jenkins-slave

```

Which makes me think the informer piece work, and that maybe my thought at https://github.com/openshift/jenkins-sync-plugin/pull/410#discussion_r779034595 is the culprit.

I'll stop here and let you pick up from there.

thanks

Reply:
hmm ... ended up having some more time today, so looked at the code some ... that Adding PodTemplate trace point seems to mean we moved past that label check I was worried about

I'll see if I can add some debug to my copy of your branch.

Reply:
OK ... shifting gears to your new e2e test case logic.  Your plugin change at least works with the master branch form of `TestConfigMapPodTemplate`

Reply:
Yeah this multi label stuff is OK too

With my suggestion at https://github.com/openshift/jenkins-sync-plugin/pull/410#pullrequestreview-845082555 and running with your branch for this things work.

Reply:
thanks @gabemontero , I just commited that fix.